common/src/Model/Brush.cpp

```diff
@@ -579,37 +509,29 @@ namespace TrenchBroom {
         }
 
         bool Brush::canMoveBoundary(const BBox3& worldBounds, const BrushFace* face, const Vec3& delta) const {
-            BrushFace* testFace = face->clone();
+            auto* testFace = face->clone();
             testFace->transform(translationMatrix(delta), false);
 
             BrushFaceList testFaces;
             testFaces.push_back(testFace);
 
-            for (BrushFace* brushFace : m_faces) {
-                if (brushFace != face)
-                    testFaces.push_back(brushFace);
-            }
-
-            BrushGeometry testGeometry(worldBounds);
-            CanMoveBoundary canMove(testGeometry, testFaces);
-            const bool inWorldBounds = worldBounds.contains(testGeometry.bounds()) && testGeometry.closed();
-
-            bool fullySpecified = true;
-            for (BrushFaceGeometry* current : testGeometry.faces()) {
-                if (current->payload() == nullptr) {
-                    fullySpecified = false;
-                    break;
+            for (auto* brushFace : m_faces) {
+                if (brushFace != face) {
+                    testFaces.push_back(brushFace->clone());
                 }
             }
 
-            restoreFaceLinks(m_geometry);
-            delete testFace;
-
-            return (fullySpecified &&
-                    inWorldBounds &&
-                    !canMove.brushEmpty() &&
-                    !canMove.hasRedundandFaces() &&
-                    !canMove.hasDroppedFaces());
+            try {
+                auto* testBrush = new Brush(worldBounds, testFaces);
+                const bool inWorldBounds = worldBounds.contains(testBrush->bounds());
+                const bool closed = testBrush->closed();
+                const bool allFaces = testBrush->faceCount() == testFaces.size();
+                delete testBrush;
+
+                return inWorldBounds && closed && allFaces;
+            } catch (...) {
```

Comment:
I worry this will catch something that shouldn't be recovered from; would catching GeometryException be enough?

Reply:
done